tests/ecmascript/test-bi-symbol-basic.js

```diff
@@ -0,0 +1,247 @@
+/*
+ *  Basic Symbol behavior.
+ */
+
+print=console.log
+
+/*===
+===*/
+
+function symbolCreationTest() {
+    var s, s1, s2;
+
+    // Constructing as 'new' is a TypeError.
+    try {
+        s = new Symbol();
+    } catch (e) {
+        print(e.name);
+    }
+    try {
+        s = new Symbol('123');
+    } catch (e) {
+        print(e.name);
+    }
+
+    // Creating an anonymous symbol.
+    s1 = Symbol();
+    s2 = Symbol();
+    print(s1);
+    print(s1.toString());
+    print(s2);
+    print(s2.toString());
+    print(s1 == s2, s1 === s2);  // never equal
+
+    // Creating a symbol with a description.
+    s1 = Symbol('123');
+    s2 = Symbol('123');
+    print(s1);
+    print(s1.toString());
+    print(s2);
+    print(s2.toString());
+    print(s1 == s2, s1 === s2);  // never equal
+
+    // Symbol() argument is string coerced.
+    s1 = Symbol(123);
+    s2 = Symbol(true);
+    print(s1);
+    print(s2);
+
+    // ToString(Symbol('foo')) is a TypeError (E6 Section 7.1.12).
+    try {
+        s1 = Symbol(Symbol('foo'));
+        print(s1);
+    } catch (e) {
+        print(e.name);
+    }
+
+    // But explicit .toString() returns e.g. "Symbol(foo)" which works as an
+    // argument for creating an odd symbol.
+    try {
+        s1 = Symbol(Symbol('foo').toString());
+        print(s1);
+    } catch (e) {
+        print(e.name);
+    }
+
+    // Missing argument and undefined are treated specially, other arguments
+    // are string coerced (even when it makes little sense).
+    s1 = Symbol();
+    s2 = Symbol(void 0);
+    print(s1);
+    print(s2);
+    s1 = Symbol(null);
+    print(s1);
+    s1 = Symbol(new ArrayBuffer(9));
+    print(s1);
+
+    // Specification differentiates between 'undefined' [[Ðescription]] and
+    // empty string description on creation, but 'undefined' description is
+    // in practice treated like an empty string where descriptions are used.
+    // FIXME: test?
+
+    //
+}
+
+try {
+    print('symbol creation');
+    symbolCreationTest();
+} catch (e) {
+    print(e.stack || e);
+}
+
+/*===
+===*/
+
+function symbolCoercionTest() {
+    var s1, s2, s3, o1, o2, o3;
+    var t;
+
+    // Object coercion creates a wrapped Symbol which non-strict equals
+    // the plain symbol.  Double Object coercion is idempotent (of course).
+    s1 = Symbol('123');
+    o1 = Object(s1);
+    print(s1 == o1, s1 === o1);
+    o2 = Object(o1);
+    print(s1 == o2, s1 === o2);
+    print(o1 === o1);
+
+    // ToString coercion is a TypeError.
+    try {
+        s1 = Symbol('123');
+        t = String(s1);
```

Comment:
Ah, my bad: in ES6 the String() constructor call has a special case for Symbols so that it doesn't end up invoking the built-in ToString() operation: http://www.ecma-international.org/ecma-262/6.0/#sec-string-constructor-string-value.

But anything that causes `ToString()` to happen should TypeError. So V8 is probably right here, Symbol() uses ToString().